STYLE.md

```diff
@@ -179,6 +179,91 @@ For words that are both concepts and API terms (for example, a `Touched` event o
 - Connect the function to the `Class.BasePart.Touched` event.
 - Display text when players touch the part.
 
+## Reference YAML Files
+
+The reference documentation in `content/en-us/reference/engine/` uses YAML rather than Markdown. YAML is a structured format that has stricter whitespace requirements than Markdown. Pay attention to the amount of leading whitespace before an element; even a single missing space can result in invalid YAML. If you're not familiar with the format, consider using an online YAML linter to check your work.
+
+The documentation makes heavy use of the `|` character, which allows for multi-line strings. It preserves line breaks and lets you include multiple paragraphs, lists, tables, etc.
+
+The reference documentation files come in five types:
+
+- Classes — core classes such as `BasePart`, `Model`, `Humanoid`, `DataStoreService`, etc.
+- Data types — data types such as `Color3`, `CFrame`, `UDim2`, `Vector3`, etc.
+- Enums — Enums such as `CameraType`, `FontSize`, `ScreenInsets`, etc.
+- Globals — Lua and Roblox globals.
+- Libraries — Lua libraries such as `math`, `string`, and `table`.
+
+Internal tooling generates the structure of class and enum YAML files. Don't add new constructors, properties, methods, or events to these files—just fill in the details.
+
+Data types, globals, and libraries do _not_ have a generated structure. If you identify a missing constructor, property, method, etc., feel free to add it. Use an existing item as a template for the missing one.
+
```

Comment:
I'm not sure about this. If done incorrectly, it might cause issues with Studio integration, which I haven't set up thorough testing for yet. Doing this will also fail the protected fields test. It's probably best to just have people edit the `summary`, `description`, and `deprecation_message` fields

Reply:
Sounds good -- updated the guidance. Let me know how it looks. 👍